HWLesson 4.6 (Module)/MigrationProject/Services/PetService.cs

```diff
@@ -0,0 +1,173 @@
+﻿using Data;
+using DTO;
+using Microsoft.Extensions.Logging;
+using Models;
+using Repositories.Abstractions;
+using Services.Abstractions;
+
+
+namespace Services
+{
+    public class PetService : BaseDataService<AppDbContext> , IPetService
+    {
+        private readonly IPetRepository _petRepository;
+        private readonly ILogger<PetService> _loggerService;
+
+        public PetService(IDbContextWrapper<AppDbContext> dbContextWrapper,
+             ILogger<BaseDataService<AppDbContext>> logger,
+             IPetRepository petRepository,
+             ILogger<PetService> loggerService)
+            : base(dbContextWrapper, logger)
+        {
+            _petRepository = petRepository;
+            _loggerService = loggerService;
+        }
+
+
+        public async Task<int> AddPetAsync(string petName, int categoryID, int breedId, float age, int locationId, string description = "", string imageURL = "")
+        {
+            var id = await _petRepository.AddPetAsync(petName, categoryID, breedId, age, locationId, imageURL, description);
+            _loggerService.LogInformation($"Created pet with id = {id}");
+            return id;
+        }
+
+        public async Task<List<Pet>> GetPetAsync()
+        {
+            var petEntity = await _petRepository.GetPetAsync();
+
+            if (petEntity.Count == 0)
+            {
+                _loggerService.LogWarning($"Pets not founded");
+                return null!;
+            }
+            else
+            {
+                List<Pet> pets = new List<Pet>();
+
+                pets?.AddRange(petEntity.Select(pet => new Pet()
+                {
+                 PetID = pet.Id,
+                 PetName = pet.Name,
+                 Category = new Category
+                 {
+                     CategoryID = pet.Category.Id,
+                     CategoryName = pet.Category.CategoryName,
+                 },
+                 Breed = new Breed() 
+                 {
+                     BreedID = pet.Breed.Id,
+                     BreedName = pet.Breed.BreedName,
+                     Category = new Category
+                     {
+                         CategoryName = pet.Breed.Category.CategoryName,
+                         CategoryID = pet.Breed.Category.Id,
+                     },
+                 },
+                 Location = new Location
+                 {
+                     Id = pet.Location.Id,
+                     LocationName = pet.Location.LocationName,
+                 },
+                 Age = pet.Age,
+                 Description = pet.Description,
+                 ImageURL = pet.ImageURL,
+                }));
+
+                return pets.OrderBy(p => p.Location.LocationName).OrderBy(p => p.Breed.Category.CategoryID).ToList();
```

Comment:
Do not use two OrderBy(), use .OrderBy(p => p.Location.LocationName).ThenBy(p => p.Breed.Category.CategoryID).ToList();